bindings/go/blob/direct/blob_test.go

```diff
@@ -183,6 +184,49 @@ func TestCompatibility(t *testing.T) {
 		assert.Equal(t, data, string(result))
 		assert.Equal(t, int64(len(data)), b.Size())
 	})
+
+	t.Run("pipe", func(t *testing.T) {
+		r, w := io.Pipe()
+		go func() {
+			_, err := w.Write([]byte(data))
+			assert.NoError(t, w.CloseWithError(err))
+		}()
+
+		b := New(r)
+		br, err := b.ReadCloser()
+
+		assert.NoError(t, err)
+		result, err := io.ReadAll(br)
+		assert.NoError(t, err)
+		assert.Equal(t, data, string(result))
+		assert.Equal(t, int64(-1), b.Size())
+		mtString, ok := b.MediaType()
+		assert.True(t, ok)
+		assert.Equal(t, "application/octet-stream", mtString)
+
+		t.Run("serialized access to pipe", func(t *testing.T) {
+			parallel := atomic.Bool{}
+			waitForParallel := make(chan struct{}, 1)
+			go func() {
+				defer func() {
+					waitForParallel <- struct{}{}
+				}()
+				// Wait for the reader to be created
+				rc, err := b.ReadCloser()
+				require.NoError(t, err)
+				require.NotNil(t, rc)
+				result, err := io.ReadAll(br)
```

Comment:
Shouldn't this be using `rc` instead of `br`?

Reply:
good catch!